core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -95,7 +94,7 @@ public ClusterState execute(ClusterState currentState) {
                         IndexRoutingTable indexRoutingTable = currentState.routingTable().index(index);
                         for (IndexShardRoutingTable shard : indexRoutingTable) {
                             for (ShardRouting shardRouting : shard) {
-                                if (shardRouting.primary() == true && shardRouting.allocatedPostIndexCreate() == false) {
+                                if (shardRouting.primary() == true && shardRouting.allocatedPostIndexCreate(indexMetaData) == false) {
```

Comment:
I think this protection can go away now? It's a protection against an index staying red when opening it (because now primary can be found). We now do a better job with the active allocation ids and we will just assign new empty primaries.